modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/QueryEntityValidationSelfTest.java

```diff
@@ -42,6 +46,31 @@
         startGrid(0);
     }
 
+    /**
+     * Create table with wrapped key and user value type and insert value by cache API.
+     * Check inserted value.
+     * @throws Exception In case of errors.
+     */
+    @Test
+    public void testWrappedKeyValidation() throws Exception {
```

Comment:
Why do you have decided to extend a `QueryEntityValidationSelfTest` with this test case?

Reply:
Move to H2DynamicTableSelfTest, but as a separate test, because it requires the value class name to reproduce.